numpy/distutils/misc_util.py

```diff
@@ -14,9 +14,15 @@
 import distutils
 from distutils.errors import DistutilsError
 try:
-    from threading import local as tlocal
+    from threading import local as tlocal, Lock
 except ImportError:
-    from dummy_threading import local as tlocal
+    from dummy_threading import local as tlocal, Lock
+
```

Comment:
Import itself is thread safe, so I assume the module executes in a single thread when imported. Would is make sense to register the cleanup function at that time?